custom/icds/rules/custom_actions.py

```diff
@@ -63,11 +63,11 @@ def _update_tech_issue_for_escalation(case, escalated_ticket_level):
     )
 
 
-def _get_escalated_tech_issue_delegate(tech_issue, escalated_location_id):
+def _get_escalated_tech_issue_delegate(tech_issue, location_id):
     for subcase in tech_issue.get_subcases(index_identifier='parent'):
         if (
             subcase.type == 'tech_issue_delegate' and
-            subcase.owner_id == escalated_location_id and
+            subcase.owner_id == location_id and
```

Comment:
This is the main issue: the code looking for an existing delegate case looks for one at the next higher level instead of the current level.